web/src/components/assets/ContractUploadForm.tsx

```diff
@@ -0,0 +1,268 @@
+        setUploadSuccess(false);
+      }, 1500);
+    } catch (error) {
+      console.error("Upload failed:", error);
+      setError("Error: " + String((error as Error).message));
+    } finally {
+      setUploading(false);
+    }
+  };
+
+  const clearFile = () => {
+    setUploadFile(null);
+    setError(null);
+    setUploadSuccess(false);
+    if (fileInputRef.current) {
+      fileInputRef.current.value = "";
+    }
+  };
+
+  const handleClose = () => {
+    reset();
+    setUploadFile(null);
+    setError(null);
+    setUploadSuccess(false);
+    if (fileInputRef.current) {
+      fileInputRef.current.value = "";
+    }
+    onClose();
+  };
+
+  if (!isOpen) return null;
+
+  return (
+    <Dialog setDialogOpen={handleClose}>
+      <h2>Upload Contract</h2>
+      <p className={styles.description}>
+        Upload a PDF contract document for this asset. Only PDF files up to 10MB are accepted.
+      </p>
+
+      <form onSubmit={handleSubmit(onSubmit)} className={styles.form}>
+        <div className={styles["upload-section"]}>
+          <div className={styles["file-input"]}>
+            <input
+              ref={fileInputRef}
+              type="file"
+              accept=".pdf,application/pdf"
+              onChange={handleFileSelect}
+              className={styles["hidden-input"]}
+              id="contract-file-input"
+            />
+            <label htmlFor="contract-file-input" className={styles["file-label"]}>
+              <div className={styles["upload-icon"]}>📄</div>
+              <span>Choose PDF file or drag and drop</span>
+            </label>
+          </div>
+
+          {uploadFile && (
+            <div className={styles["selected-file"]}>
+              <div className={styles["file-info"]}>
+                <span className={styles["file-name"]}>{uploadFile.name}</span>
+              </div>
+              <Button onClick={clearFile} size="small" variant="tertiary">
+                Remove
+              </Button>
+            </div>
+          )}
+
+          {error && <div className={styles.error}>{error}</div>}
+
+          {uploadSuccess && <div className={styles.success}>Contract uploaded successfully!</div>}
+        </div>
+
+        <div className={styles["form-section"]}>
+          <h4>Contract Details</h4>
+
+          <div className={styles["form-group"]}>
+            <label htmlFor="organisationSignatory">Organisation Signatory *</label>
+            <input
+              id="organisationSignatory"
+              type="text"
+              {...register("organisationSignatory", {
+                required: "Organisation Signatory is required",
+                minLength: { value: 2, message: "Organisation Signatory must be at least 2 characters" },
+                maxLength: { value: 100, message: "Organisation Signatory must be less than 100 characters" },
+              })}
+              className={styles["form-input"]}
+              placeholder="Enter organisation signatory name"
```

Comment:
might be worth adding an eg. I'm not sure if this is meant to be a person's name at the org or the org itself

Reply:
> but is it in principle possible to construct queries to retrieve all contracts and assets associated with Studies that a particular IAO owns?

yeah, no sweat

Reply:
> but is it in principle possible to construct queries to retrieve all contracts and assets associated with Studies that a particular IAO owns

Yes we can do this. Do you need this to be built into this PR (will need UI/UX considerations). Or is this something just to keep in mind for future functionality?